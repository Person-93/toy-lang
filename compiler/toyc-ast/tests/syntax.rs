use anyhow::Result;
use chumsky::{Parser, Stream};
use logos::Logos;
use test_util::{assert_debug_snapshot, snapshots};
use toyc_ast::{ast, tokens::Token};

fn main() -> Result<()> {
  snapshots(|text| {
    let tokens = Token::lexer(text).spanned();
    assert_debug_snapshot!(
      ast::parse::file().parse_recovery(Stream::from_iter(text.len()..text.len(), tokens))
    );
  })
}
