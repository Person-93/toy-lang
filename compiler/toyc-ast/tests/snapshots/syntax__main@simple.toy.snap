---
source: compiler/toyc-ast/tests/syntax.rs
assertion_line: 10
expression: "ast::parse::file().parse_recovery(Stream::from_iter(text.len()..text.len(),\n                                                    tokens))"

---
(
    Some(
        [
            TopLevel {
                vis: None,
                item: Function(
                    Function {
                        extern_: None,
                        ident: Ident(f),
                        generic_params: None,
                        params: [],
                        return_type: None,
                        body: Some(
                            Expr {
                                primary: CodeBlock(
                                    CodeBlock {
                                        statement: [],
                                        expr: None,
                                    },
                                ),
                                args: None,
                            },
                        ),
                    },
                ),
            },
        ],
    ),
    [],
)
